### ISSUE TITLE
accessibility issue with LinkCard

### ISSUE THREAD

**Issue by mayank99** [withastro/starlight#487]
### What version of `starlight` are you using?

0.7.0

### What version of `astro` are you using?

2.10.4

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

Firefox, Chrome, Safari

### Describe the Bug

Love the addition of the `LinkCard` component, but it includes the entire contents of the card as part of the link, instead of just the title. This results in verbose announcements by screen readers, when really only the title should be announced.

To fix the markup but keep the whole card clickable, a pseudo-element can be used.
```html
<div class=card>
  <a href="/">the title</a>
  <div>the other stuff</div>
</div>
```
```css
a::before {
  content: '';
  position: absolute;
  inset: 0;
}
```

Further reading: https://inclusive-components.design/cards/

Happy to contribute if this is something desirable.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for the issue @mayank99! I agree! I did consider a design like that during the PR review, but felt we might just be on the terse enough side to get away with a single link. Would be happy for a PR implementing the approach you sketch out though. 💜
